### ISSUE TITLE
SQLiteException: duplicate column name: Id

### ISSUE THREAD

**Issue by MrTony1987** [gilzoide/unity-sqlite-net#30]
```
 void Start()
    {
        // 1. Create a connection to the database.
        // The special ":memory:" in-memory database and
        // URIs like "file:///somefile" are also supported
        var db = new SQLiteConnection($"{Application.persistentDataPath}/MyDb.db");

        // 2. Once you have defined your entity, you can automatically
        // generate tables in your database by calling CreateTable
        db.CreateTable<Player>();

        // 3. You can insert rows in the database using Insert
        // The Insert call fills Id, which is marked with [AutoIncremented]
        var newPlayer = new Player
        {
            Name = "gilzoide",
        };
        Debug.Log("before insert Id=" + newPlayer.Id);
        db.Insert(newPlayer);
        Debug.Log("after insert Id=" + newPlayer.Id);
        Debug.Log($"Player new ID: {newPlayer.Id}");
        // Similar methods exist for Update and Delete.

        // 4.a The most straightforward way to query for data
        // is using the Table method. This can take predicates
        // for constraining via WHERE clauses and/or adding ORDER BY clauses
        var query = db.Table<Player>().Where(p => p.Name.StartsWith("g"));
        foreach (Player player in query)
        {
            Debug.Log($"Found player named {player.Name} with ID {player.Id}");
        }

        // 4.b You can also make queries at a low-level using the Query method
        var players = db.Query<Player>("SELECT * FROM Player WHERE Id = ?", 1);
        foreach (Player player in players)
        {
            Debug.Log($"Player with ID 1 is called {player.Name}");
        }

        // 5. You can perform low-level updates to the database using the Execute
        // method, for example for running PRAGMAs or VACUUM
        db.Execute("VACUUM");
    }
```

Hello, I found a problem. After building and running the above example on an Android device, starting from the second launch, an exception is thrown: “SQLiteException: duplicate column name: Id.”. 
Editor is OK.   v1.1.0

**Comment by gilzoide** [comment]
Hey @MrTony1987, thanks for the report.
That's really weird 🤔 How is your "Player" class defined?

**Comment by MrTony1987** [comment]
> Hey @MrTony1987, thanks for the report. That's really weird 🤔 How is your "Player" class defined?

Hello, my code only contains the content in the Start method, with nothing else. You can freely create a MonoBehaviour script that includes this Start method and attach it to a GameObject. Then, print the log to the screen, and run it multiple times on an Android device to check the logs.

**Comment by gilzoide** [comment]
Well, you didn't really put the definition of Player in your code in this issue. I'm assuming you are using the same code that is present in this repository's README file.
```cs
public class Player
{
    [PrimaryKey, AutoIncrement]
    public int Id { get; set; }
    public string Name { get; set; }
}
```

I'll try to reproduce on an Android build here.

**Comment by gilzoide** [comment]
Also, tip for next time: use "```cs" to highlight code blocks with C# syntax =]

**Comment by longnvgeargames** [comment]
Is there any update on this? I also face the same problem.

**Comment by gilzoide** [comment]
No updates from me, at least. In my latest test, I got the "Id" column in the "CREATE TABLE" statement by lowering the managed code stripping level. It's possible that the attributes are being stripped from the builds, we can try to preserve them or something like that (reference documentation [here](https://docs.unity3d.com/Manual/managed-code-stripping.html)).
Unfortunately I won't have time to look into this for a couple of weeks.

**Comment by ZezhongWang** [comment]
I had the same issue with the iOS platform also. 

```
SQLiteException: duplicate column name: LevelName 
  at SQLite.SQLite3.Prepare2 (System.IntPtr db, System.String query) [0x00000] in <00000000000000000000000000000000>:0  
  at SQLite.SQLiteCommand.ExecuteNonQuery () [0x00000] in <00000000000000000000000000000000>:0  
  at SQLite.SQLiteConnection.Execute (System.String query, System.Object[] args) [0x00000] in <00000000000000000000000000000000>:0  
  at SQLite.SQLiteConnection.MigrateTable (SQLite.TableMapping map, System.Collections.Generic.List`1[T] existingCols) [0x00000] in <00000000000000000000000000000000>:0  
  at SQLite.SQLiteConnection.CreateTable (System.Type ty, SQLite.CreateFlags createFlags) [0x00000] in <00000000000000000000000000000000>:0  
  at GameDataManager.LoadDataModel () [0x00000] in <00000000000000000000000000000000>:0  
  at GameDataManager.Awake () [0x00000] in <00000000000000000000000000000000>:0  
```

The table structure like this:
```
public class WaveDataModel
{
    // Remove the static dbPath and replace it with a method
    private static string dbFileName = "EnemyWave.db";

    public static string GetDbPath()
    {
        string dbPath = Path.Combine(Application.streamingAssetsPath, dbFileName);
        return dbPath;
    }

    public string LevelName { get; set; }

    public int WaveNumber { get; set; }
}
```

Loading logic:

```
    private void LoadDataModel()
    {
        // Open a connection to the SQLite database
        using (SQLiteConnection db = new SQLiteConnection(WaveDataModel.GetDbPath()))
        {
            var result = db.CreateTable<WaveDataModel>();
            WaveDataArray = db.Table<WaveDataModel>().ToArray();

            string LevelName = SceneManager.GetActiveScene().name;
            WaveDataArray = WaveDataArray.Where(wave => wave.LevelName == LevelName).ToArray();
            Debug.LogFormat("Load enemy wave done, {0} waves in total.", WaveCount);
        }
    }
```

**Comment by gilzoide** [comment]
Ok, found the culprit. Unity managed code stripping is removing the `SQLite.SQLiteConnection.ColumnInfo.Name` property setter, which causes SQLite-net to ignore it as a valid and thus always returning `null` for column names. This affects the ORM in several places.

Adding a `[Preserve]` or `[RequiredMember]` attribute to this property fixes the issue. I'll open a PR soon with the fix and release the fixed version, I'll keep you posted.

**Comment by gilzoide** [comment]
There it is, this issue was fixed and is available in version 1.1.1

Thanks again for the bug report!
Cheers \o/
